src/inc/coredistools.h

```diff
@@ -0,0 +1,155 @@
+typedef bool(*OffsetComparator)(const void *UserData, size_t BlockOffset,
+    size_t InstructionLength, uint64_t Offset1,
+    uint64_t Offset2);
+
+// The Export/Import definitions for CoreDistools library are defined below.
+// A typedef for a pointer to each interface function is also defined
+// in order to aid the importer.
+
+// Initialize the disassembler, using default print controls
+DllIface CorDisasm *InitDisasm(enum TargetArch Target);
+typedef CorDisasm *InitDisasm_Sig(enum TargetArch Target);
+
+// Initialize the disassembler using custom print controls
+DllIface CorDisasm *NewDisasm(enum TargetArch Target,
+    const PrintControl *PControl);
+typedef CorDisasm *NewDisasm_Sig(enum TargetArch Target,
+    const PrintControl *PControl);
+
+// Delete the disassembler
+DllIface void FinishDisasm(const CorDisasm *Disasm);
+typedef void FinishDisasm_Sig(const CorDisasm *Disasm);
+
+// DisasmInstruction -- Disassemble one instruction
+// Arguments:
+// Disasm -- The Disassembler
+// Address -- The address at which the bytes of the instruction
+//            are intended to execute
+// Bytes -- Pointer to the actual bytes which need to be disassembled
+// MaxLength -- Number of bytes available in Bytes buffer
+// Returns:
+//   -- The Size of the disassembled instruction
+//   -- Zero on failure
+DllIface size_t DisasmInstruction(const CorDisasm *Disasm,
+    const uint8_t *Address, const uint8_t *Bytes,
+    size_t Maxlength);
+typedef size_t DisasmInstruction_Sig(const CorDisasm *Disasm,
+    const uint8_t *Address,
+    const uint8_t *Bytes, size_t Maxlength);
+
+// Initialize the Code Differ
+DllIface CorAsmDiff *NewDiffer(enum TargetArch Target,
+    const PrintControl *PControl,
+    const OffsetComparator Comparator);
+typedef CorAsmDiff *NewDiffer_Sig(enum TargetArch Target,
+    const PrintControl *PControl,
+    const OffsetComparator Comparator);
+
+// Delete the Code Differ
+DllIface void FinishDiff(const CorAsmDiff *AsmDiff);
+typedef void FinishDiff_Sig(const CorAsmDiff *AsmDiff);
+
+// NearDiffCodeBlocks -- Compare two code blocks for semantic
+//                       equivalence
+// Arguments:
+// AsmDiff -- The Asm-differ
+// UserData -- Any data the user wishes to pass through into
+//             the OffsetComparator
+// Address1 -- Address at which first block will execute
+// Bytes1 -- Pointer to the actual bytes of the first block
+// Size1 -- The size of the first block
+// Address2 -- Address at which second block will execute
+// Bytes2 -- Pointer to the actual bytes of the second block
+// Size2 -- The size of the second block
+// Returns:
+//   -- true if the two blocks are equivalent, false if not.
+DllIface bool NearDiffCodeBlocks(const CorAsmDiff *AsmDiff,
+    const void *UserData, const uint8_t *Address1,
+    const uint8_t *Bytes1, size_t Size1,
+    const uint8_t *Address2, const uint8_t *Bytes2,
+    size_t Size2);
+typedef bool NearDiffCodeBlocks_Sig(const CorAsmDiff *AsmDiff,
```

Comment:
I realize I missed this before, but I prefer a _t suffix over _Sig, but it's not a big deal.  Also, you can use the signature type to declare the function like "DllIFace NearDifferCodeBlocks_Sig NearDifferCodeBlocks;" to save duplication.  You don't have to change either of these, though.